Cannot compile the example in the project

$ cargo build --example rtic
   Compiling atat v0.11.1 (C:\Users\x\tmp\atat\atat)
error[E0433]: failed to resolve: use of undeclared crate or module `hal`
  --> atat\examples\rtic.rs:23:16
   |
23 | #[app(device = hal::pac, peripherals = true)]
   |                ^^^ use of undeclared crate or module `hal`

error[E0423]: expected function, tuple struct or tuple variant, found struct `Queue`
  --> atat\examples\rtic.rs:32:47
   |
32 |         static mut RES_QUEUE: ResQueue<256> = Queue(heapless::i::Queue::u8());
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |



rustc 1.53.0 (53cb7b09b 2021-06-17)
cargo 1.53.0 (4369396ce 2021-04-27)

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -110,6 +110,8 @@ jobs:
           - target: "x86_64-unknown-linux-gnu"
             features: "derive, defmt-default"
             std: ", std"
+          - target: "thumbv7m-none-eabi"
+            examples: "--examples"
     steps:
       - name: Checkout source code
         uses: actions/checkout@v2
@@ -126,7 +128,7 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --all --target '${{ matrix.target }}' --features '${{ matrix.features }}${{ matrix.std }}'
+          args: --all --target '${{ matrix.target }}' --features '${{ matrix.features }}${{ matrix.std }}' ${{ matrix.examples }}
 
   test:
     name: Test

--- a/atat/Cargo.toml
+++ b/atat/Cargo.toml
@@ -32,7 +32,10 @@ cortex-m = "0.7.1"
 cortex-m-rt = "0.6.13"
 cortex-m-rtic = "0.5.5"
 panic-halt = "0.2.0"
-stm32l4xx-hal = { version = "0.6", features = ["stm32l4x5", "rt"] }
+stm32l4xx-hal = { git = "https://github.com/MathiasKoch/stm32l4xx-hal", branch = "enhancement/prepare-embedded-hal-1.0", features = [
+    "stm32l4x5",
+    "rt",
+] }
 
 [features]
 default = ["derive"]

--- a/atat/examples/cortex-m-rt.rs
+++ b/atat/examples/cortex-m-rt.rs
@@ -13,16 +13,15 @@ use hal::{
     timer::{Event, Timer},
 };
 
-use atat::{
-    digest::DefaultDigester, urc_matcher::DefaultUrcMatcher, ClientBuilder, ComQueue, Queues,
-    ResQueue, UrcQueue,
-};
+use atat::{AtatClient, ClientBuilder, ComQueue, Queues, ResQueue, UrcQueue};
 
-use heapless::{consts, spsc::Queue};
+use heapless::spsc::Queue;
 
-use crate::rt::entry;
+use cortex_m_rt::entry;
 
-static mut INGRESS: Option<atat::IngressManager<256>> = None;
+static mut INGRESS: Option<
+    atat::IngressManager<atat::DefaultDigester, atat::DefaultUrcMatcher, 256, 10>,
+> = None;
 static mut RX: Option<Rx<USART2>> = None;
 
 #[entry]
@@ -65,9 +64,9 @@ fn main() -> ! {
 
     serial.listen(Rxne);
 
-    static mut RES_QUEUE: ResQueue<256> = Queue(heapless::i::Queue::u8());
-    static mut URC_QUEUE: UrcQueue<256, 10> = Queue(heapless::i::Queue::u8());
-    static mut COM_QUEUE: ComQueue = Queue(heapless::i::Queue::u8());
+    static mut RES_QUEUE: ResQueue<256> = Queue::new();
+    static mut URC_QUEUE: UrcQueue<256, 10> = Queue::new();
+    static mut COM_QUEUE: ComQueue = Queue::new();
 
     let queues = Queues {
         res_queue: unsafe { RES_QUEUE.split() },
@@ -93,10 +92,10 @@ fn main() -> ! {
         asm::wfi();
 
         match client.send(&common::AT) {
-            Ok(response) => {
+            Ok(_response) => {
                 // Do something with response here
             }
-            Err(e) => {}
+            Err(_e) => {}
         }
     }
 }
@@ -111,7 +110,12 @@ fn TIM7() {
 fn USART2() {
     let ingress = unsafe { INGRESS.as_mut().unwrap() };
     let rx = unsafe { RX.as_mut().unwrap() };
-    if let Ok(d) = nb::block!(rx.read()) {
+    if let Ok(d) = nb::block!(rx.try_read()) {
         ingress.write(&[d]);
     }
+}
+
+#[panic_handler] // panicking behavior
+fn panic(_info: &core::panic::PanicInfo) -> ! {
+    cortex_m::peripheral::SCB::sys_reset();
 }

--- a/atat/examples/rtic.rs
+++ b/atat/examples/rtic.rs
@@ -1,26 +1,21 @@
 #![no_main]
 #![no_std]
 
-use stm32l4xx_hal as hal;
-
 mod common;
 
-use hal::{
+use stm32l4xx_hal::{
     pac::{Peripherals, USART2},
     prelude::*,
     serial::{Config, Event::Rxne, Rx, Serial},
     timer::Timer,
 };
 
-use atat::{
-    digest::DefaultDigester, urc_matcher::DefaultUrcMatcher, ClientBuilder, ComQueue, Queues,
-    ResQueue, UrcQueue,
-};
+use atat::{AtatClient, ClientBuilder, ComQueue, Queues, ResQueue, UrcQueue};
 use rtic::{app, export::wfi};
 
 use heapless::spsc::Queue;
 
-#[app(device = hal::pac, peripherals = true)]
+#[app(device = stm32l4xx_hal::pac, peripherals = true)]
 const APP: () = {
     struct Resources {
         ingress: atat::IngressManager<atat::DefaultDigester, atat::DefaultUrcMatcher, 256, 10>,
@@ -29,9 +24,9 @@ const APP: () = {
 
     #[init(spawn = [at_loop])]
     fn init(ctx: init::Context) -> init::LateResources {
-        static mut RES_QUEUE: ResQueue<256> = Queue(heapless::i::Queue::u8());
-        static mut URC_QUEUE: UrcQueue<256, 10> = Queue(heapless::i::Queue::u8());
-        static mut COM_QUEUE: ComQueue = Queue(heapless::i::Queue::u8());
+        static mut RES_QUEUE: ResQueue<256> = Queue::new();
+        static mut URC_QUEUE: UrcQueue<256, 10> = Queue::new();
+        static mut COM_QUEUE: ComQueue = Queue::new();
 
         let p = Peripherals::take().unwrap();
 
@@ -58,7 +53,7 @@ const APP: () = {
         let tx = gpioa.pa2.into_af7(&mut gpioa.moder, &mut gpioa.afrl);
         let rx = gpioa.pa3.into_af7(&mut gpioa.moder, &mut gpioa.afrl);
 
-        let mut timer = Timer::tim7(p.TIM7, 1.hz(), clocks, &mut rcc.apb1r1);
+        let timer = Timer::tim7(p.TIM7, 1.hz(), clocks, &mut rcc.apb1r1);
 
         let mut serial = Serial::usart2(
             p.USART2,
@@ -71,9 +66,9 @@ const APP: () = {
         serial.listen(Rxne);
 
         let queues = Queues {
-            res_queue: unsafe { RES_QUEUE.split() },
-            urc_queue: unsafe { URC_QUEUE.split() },
-            com_queue: unsafe { COM_QUEUE.split() },
+            res_queue: RES_QUEUE.split(),
+            urc_queue: URC_QUEUE.split(),
+            com_queue: COM_QUEUE.split(),
         };
 
         let (tx, rx) = serial.split();
@@ -82,7 +77,7 @@ const APP: () = {
 
         ctx.spawn.at_loop().unwrap();
 
-        let response = client.send(&common::AT).unwrap();
+        client.send(&common::AT).unwrap();
 
         init::LateResources { ingress, rx }
     }
@@ -106,9 +101,9 @@ const APP: () = {
     }
 
     #[task(binds = USART2, priority = 4, resources = [ingress, rx])]
-    fn serial_irq(mut ctx: serial_irq::Context) {
+    fn serial_irq(ctx: serial_irq::Context) {
         let rx = ctx.resources.rx;
-        if let Ok(d) = nb::block!(rx.read()) {
+        if let Ok(d) = nb::block!(rx.try_read()) {
             ctx.resources.ingress.write(&[d]);
         }
     }
@@ -119,3 +114,8 @@ const APP: () = {
         fn LCD();
     }
 };
+
+#[panic_handler] // panicking behavior
+fn panic(_info: &core::panic::PanicInfo) -> ! {
+    cortex_m::peripheral::SCB::sys_reset();
+}

